### ISSUE TITLE
How to visualize crossattention and selfattention?

### ISSUE THREAD

**Issue by superhero-7** [rohitgandikota/erasing#3]
Hi! Thanks for your excellent work!

The visualization of the crossattention and selfattention in your project page looks great! Like this:
![image](https://user-images.githubusercontent.com/57797766/226156781-bbb301e0-9615-49d2-b9d5-38683d18bfde.png)

And I am courious how to visualize it? Could you provide some example code  or some suggestion of implementing the visualization? 

By the way, what is the meaning of "activate" meantioned in your project page? I am looking forward your reply. Thanks in first!

**Comment by rohitgandikota** [comment]
Hi, thanks!

For the visualization, we simply placed forward hooks on the attention layer outputs and collected the raw activations of the layers. Then we just extrapolated them to desired image size and displayed them (just the raw activations).

By active, we mean that the attention (softmax output) attend to the image. In other words, their activation is high (orange color in the heat map shown above).

**Comment by superhero-7** [comment]
ok！Thank your reply! It's more clear right now.
But what exactly I want to know is how to extropolate the attention map size (e.g. [batch_size, 4096, 320]) to desired image size (e.g. [batch_size, 512*512, 3])?
Could you give some example code? Thanks very much!

**Comment by superhero-7** [comment]
ok！Thank your reply! It's more clear right now.
But what exactly I want to know is how to extropolate the attention map size (e.g. [batch_size, 4096, 77]) to desired image size (e.g. [batch_size, 512*512, 3])?
Could you give some example code? Thanks very much!

**Comment by rohitgandikota** [comment]
[batch_size, 4096, 77] if we choose the token_id that we want to see in cross attentions (the last dim) we get [batch_size, 4096], reshape it to [batch_size, 64, 64] and then simply resize.

**Comment by superhero-7** [comment]
Just resize [batch_size, 64, 64] to [batch_size, 512, 512] is ok? Like torchvision's resize function?

**Comment by rohitgandikota** [comment]
torch.nn.functional.interpolate this should do the job
